### PR TITLE
Move config documentation into docstrings

### PR DESCRIPTION
**Pull request by Ayushsunny**
Move config documentation into docstrings
issue code - #916

### CODE REVIEW

**Comment by Ayushsunny** [comment]
Hello @vringar could you review this PR that I'm doing right or not? please

File: openwpm/config.py [hunk 1]
```diff
@@ -116,19 +116,44 @@ class ManagerParams(DataClassJsonMixin):
     run
     """
 
-    data_directory: Path = field(
+    data_directory:
+    """The directory into which screenshots and page dumps will be saved"""
+     Path = field(
         default=Path.home() / "openwpm",
         metadata=DCJConfig(encoder=path_to_str, decoder=str_to_path),
     )
```
Comment (vringar) [inline openwpm/config.py:124]:
```suggestion
    data_directory:Path = field(
        default=Path.home() / "openwpm",
        metadata=DCJConfig(encoder=path_to_str, decoder=str_to_path),
    )
    """The directory into which screenshots and page dumps will be saved"""
```

**Review by vringar — CHANGES_REQUESTED** [review]
Hey @Ayushsunny,
sorry for taking a while to get back to you.
I left a comment showing how the docstring should be formatted.

**Comment by vringar** [comment]
As you can see that all the tests are currently failing because where you inserted the docstrings are invalid python.

File: openwpm/config.py (hunk 1 above)
Comment (laudalehsan) [reply openwpm/config.py:124]: Thank you so much sir.

Comment (Ayushsunny) [reply openwpm/config.py:124]: Thanks @vringar

**Comment by vringar** [comment]
This looks really good, thank you!
Could you please also remove the lines from Configuration.md.

**Comment by Ayushsunny** [comment]
> This looks really good, thank you!
> Could you please also remove the lines from Configuration.md.

@vringar Done

File: openwpm/config.py [hunk 2]
```diff
@@ -120,15 +120,34 @@ class ManagerParams(DataClassJsonMixin):
...
+    """The path to the file in which OpenWPM will log. The
+    directory given will be created if it does not exist."""
     testing: bool = False
+    """A platform wide flag that can be used to only run certain functionality
+    while testing. For example, the Javascript instrumentation"""
     memory_watchdog: bool = False
+    """- It is part of default manager_params. It is set to false by default which can manually be set to true.
+       - A watchdog that tries to ensure that no Firefox instance takes up too much memory. It is set to false by default
```
Comment (vringar) [inline openwpm/config.py:135]: Same here.

File: openwpm/config.py [hunk 3]
```diff
@@ -120,15 +120,34 @@ class ManagerParams(DataClassJsonMixin):
...
+       - It is mostly useful for long running cloud crawls"""
     process_watchdog: bool = False
+    """- It is part of default manager_params. It is set to false by default which can manually be set to true.
```
Comment (vringar) [inline openwpm/config.py:138]: And here.

File: openwpm/config.py [hunk 4]
```diff
@@ -120,15 +120,34 @@ class ManagerParams(DataClassJsonMixin):
...
+       - It is used to create another thread that kills off `GeckoDriver` (or `Xvfb`) instances that haven't been spawned by OpenWPM. (GeckoDriver is used by
+         Selenium to control Firefox and Xvfb a "virtual display" so we simulate having graphics when running on a server)."""
     num_browsers: int = 1
     _failure_limit: Optional[int] = None
+    """has to be either of type `int` or `None`
```
Comment (vringar) [inline openwpm/config.py:143]: The type is also captured in the line above.

Comment (Ayushsunny) [reply openwpm/config.py]: Removed

**Comment by Ayushsunny** [comment]
Can you see now @vringar ?

File: openwpm/config.py [hunk 5]
```diff
@@ -120,15 +120,30 @@ class ManagerParams(DataClassJsonMixin):
...
         default=Path.home() / "openwpm" / "openwpm.log",
         metadata=DCJConfig(encoder=path_to_str, decoder=str_to_path),
     )
+    """The path to the file in which OpenWPM will log. The
+    directory given will be created if it does not exist."""
     testing: bool = False
+    """A platform wide flag that can be used to only run certain functionality
+    while testing. For example, the Javascript instrumentation"""
     memory_watchdog: bool = False
+    """- It is mostly useful for long running cloud crawls"""
```
Comment (vringar) [inline openwpm/config.py:134]:
```suggestion
    """A watchdog that tries to ensure that no Firefox instance takes up too much memory. It is mostly useful for long running cloud crawls"""
```
Sorry for being imprecise with my feedback.

**Review by vringar — APPROVED** [review]

File: openwpm/config.py (hunk 5 above)
Comment (vringar) [reply openwpm/config.py]: I'm very happy you are getting something out of this as well!

**Comment by Ayushsunny** [comment]
Hello @vringar Can I work on some other issue's if you think I'm eligible to do?

**Pull request merged**